src/cv/apply_detections.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+import argparse
+import imutils
+import json
+import pickle
+import cv2
+import os
+import settings
+import time
+import yolo_video_detect as obj_detector
+import recognize_faces as face_recognize
+#import cv_producer
+#import insert_frame
+
+def face_tracker_helper(face_tracker, new_detections):
+    """ Returns True if there is at least 1 new detection """
+    ret_val = False
+    for i in range(len(new_detections)):
+        if new_detections[i] == "unknown":
+            continue
+        elif not new_detections[i] in face_tracker:
+            face_tracker.append(new_detections[i])
+            ret_val = True
+    return ret_val
+
+def init_video_stream(args):
+    """ Obtains the video stream """
```

Comment:
Replace 'obtain' with return

What happens if the stream doesn't exist? Did you put guards around your code if it doesn't exist?

Reply:
It won't be able read a frame from the stream so it will break out of loop there.